lib/chef/resource/file/verification/json.rb

```diff
@@ -0,0 +1,52 @@
+#
+# Author:: Antony Thomas (<antonydeepak@gmail.com>)
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "json" unless defined?(JSON)
+
+class Chef
+  class Resource
+    class File
+      class Verification
+
+        #
+        # Extends File verification to provide Json verification
+        #
+        # Example:
+        # file 'foo.json' do
+        #   content '{"foo": "bar"}'
+        #   verify :json
+        # end
+        #
+        #
+
+        class Json < Chef::Resource::File::Verification
+
+          provides :json
+
+          def verify(path, opts = {})
+            JSON.parse(IO.read(path))
```

Comment:
This should probably use Chef::JSONCompat instead of the JSON gem.

Reply:
I swapped to using `Chef::JSONCompat` however the expectation for spec "returns false for empty file" fails. Looks like the yajl parser wrapped by compat does not seem to comply with Json grammar for empty string?

Reply:
Thought about this a bit more and changed the behavior of spec for reasons documented [here](https://github.com/chef/chef/pull/12062/files#diff-94074dd14d988f4d99ec8263cb93dbafda6cb5f1f91b3fd0168a79ffe854531fR55). lmk what you think.

Reply:
yeah i think its better to just merge this and ship it.  the comment there will be good if we need to revisit.